app/src/main/java/com/gmail/sanovikov71/githubclient/ui/list/MainActivity.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    @Override
+    public void onRefresh() {
+        if (null != mDataService) {
+            showProgressDialog();
+            mDataService.fetchUsers(this);
+        }
+    }
+
+    @Override
+    public void loadMore() {
+        mDataService.fetchMoreUsers(this, mNumberOfLoadedEntries);
+    }
+
+    @Override
+    public void onClickUser(int userId) {
+        updateRecentsStorage(userId);
+        Intent intent = new Intent(this, DetailActivity.class);
+        intent.putExtra(DetailActivity.EXTRA_USER_ID, userId);
+        startActivity(intent);
+    }
+
+    private void updateRecentsStorage(int newRecentUserId) {
+
+        final String newRecentId = String.valueOf(newRecentUserId);
+
+        SharedPreferences recents =
+                getSharedPreferences(RecentsFragment.RECENT_LIST_PREFERENCES_NAME, MODE_PRIVATE);
+
+        // format: 1;7;8;3;12;
+        String recentsStr = recents.getString(RecentsFragment.KEY_RECENTS, "");
+
+        List<String> recentList = RecentsFragment.getRecentUsers(recentsStr);
+
+        if (newRecentId.equals(recentList.get(0))) {
+            return;
+        }
+
+        int indexOf = recentList.indexOf(newRecentId);
+        if (-1 != indexOf) {
+            recentList.remove(indexOf);
+        } else {
+            recentList.remove(recentList.size() - 1);
+        }
+        recentList.add(0, newRecentId);
+
+        // save changes
+        StringBuilder newRecentsStrBuilder = new StringBuilder();
+        for (int i = 0; i < recentList.size(); i++) {
+            newRecentsStrBuilder.append(recentList.get(i)).append(RecentsFragment.RECENT_SEPARATOR);
+        }
+        recents.edit().putString(RecentsFragment.KEY_RECENTS, newRecentsStrBuilder.toString()).apply();
+
+    }
+
+    @Override
+    public void onSearchResultShown() {
+        hideRecentsFragment();
+    }
+
+    @Override
+    public void onNoSearchResults() {
+        showRecentsFragment();
+    }
+
+    @Override
+    public void onSearchQuerySubmitted(String query) {
+        mDataService.searchUsers(this, query);
+    }
+
+    private void showRecentsFragment() {
+        getSupportFragmentManager()
+                .beginTransaction()
+                .replace(R.id.recents_fragment_container, new RecentsFragment(), TAG_RECENTS_FRAGMENT)
+                .commit();
+        mRecentContainer.setVisibility(View.VISIBLE);
+    }
+
+    private void hideRecentsFragment() {
+        final FragmentManager supportFragmentManager = getSupportFragmentManager();
+        Fragment recentsFragment = supportFragmentManager.findFragmentByTag(TAG_RECENTS_FRAGMENT);
+        supportFragmentManager.beginTransaction().remove(recentsFragment).commit();
+        mRecentContainer.setVisibility(View.GONE);
+    }
+
+    @Override
+    public void onSearchResult(List<User> searchResult) {
+        ((SearchFragment) getSupportFragmentManager().findFragmentById(R.id.search_fragment))
+                .searchResultsLoaded(searchResult);
```

Comment:
First of all FragmentManager can return null or instance of another class, second of all you should implement something better than calling public methods to make communications between your modules.